internal/shared/editor_helper.go

```diff
@@ -0,0 +1,49 @@
+package shared
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+var editor = "vim"
+
+func init() {
+	if s := os.Getenv("VISUAL"); s != "" {
+		editor = s
+	} else if s := os.Getenv("EDITOR"); s != "" {
+		editor = s
+	}
+}
+
+func OpenDefaultEditor(data []byte) ([]byte, error) {
+	return openEditor(editor, data)
+}
+
+func openEditor(editor string, data []byte) ([]byte, error) {
+	tempFile, err := os.CreateTemp("", "redit")
+	if err != nil {
+		return nil, fmt.Errorf("os.CreateTemp: %w", err)
+	}
+	defer os.Remove(tempFile.Name())
+
+	if _, err := tempFile.Write(data); err != nil {
```

Comment:
I think this was previously enclosed in an `if len(data) != 0`. But that test is unnecessary. If `len(data)` is 0, the call to `tempFile.Write` will just be a no-op.